File: src/main/java/org/orbeon/saxon/dom/TypedDocumentWrapper.java
package org.orbeon.saxon.dom;

import org.orbeon.dom.Document;
import org.orbeon.dom.Node;
import org.orbeon.saxon.Configuration;

/**
 * This wrapper is an extension of the Saxon node wrapper which is aware of XForms type annotations.
 */
public class TypedDocumentWrapper extends DocumentWrapper {

    public TypedDocumentWrapper(Document document, String s, Configuration configuration) {
        super(document, s, configuration);
    }

    @Override
    protected NodeWrapper makeWrapper(Node node, DocumentWrapper docWrapper, NodeWrapper parent, int index) {
        return TypedNodeWrapper.makeTypedWrapper(node, docWrapper, parent, index);
    }
}

File: xforms-runtime/js/src/main/scala/org/orbeon/dom/saxon/TypedDocumentWrapper.scala
package org.orbeon.dom.saxon

import org.orbeon.dom
import org.orbeon.oxf.util.StaticXPath.SaxonConfiguration

/**
 * This wrapper is an extension of the Saxon node wrapper which is aware of XForms type annotations.
 */
class TypedDocumentWrapper(
  val document      : dom.Document,
  val s             : String,
  val configuration : SaxonConfiguration
) extends DocumentWrapper(document, s, configuration) {

  override protected def makeWrapper(
    node       : dom.Node,
    docWrapper : DocumentWrapper,
    parent     : NodeWrapper
  ): NodeWrapper =
    TypedNodeWrapper.makeTypedWrapper(node, docWrapper, parent)
}